google/api/http.proto

```diff
@@ -346,6 +339,18 @@ message HttpRule {
   // message type.
   string body = 7;
 
+  // The name of fields whose values are mapped to HTTP headers in a request. If
+  // a header has a case-insensitive match to a key in this map, the field in
+  // the request that matches the value will have it's value set with the value
+  // in the header.
+  //
+  // If a field is specified in this map, it will not be mapped to a query
+  // parameter.
```

Comment:
The current implementation already allows a parameter be duplicated by body and HTTP query parameters. We cannot suddenly block a parameter from query parameter. That is a break changing.

Reply:
If an existing field was added to this map, then yes, I think that would be a breaking change. I imagined that, even once this mapping exists, then existing fields would not map to it.

I don't think the alternative is desirable: that would imply that any field not in the body would always be able to be provided as a query parameter, and then optionally a header parameter (by adding it to the map).

There would be no way to specify that a field is exclusively honored in the header.

Reply:
will not be => must not be

Note: This doesn't work for Google APIs. Every Google API request field can be specified by an HTTP request parameter, regardless whether it is mentioned by google.api.http. Please verify other implementations of gRPC transcoding. The current implementation is copy every value from request parameters into body blindly.